src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -785,6 +790,7 @@ namespace
                 {
                     create_object_instance(
                         object_assembly.ref(),
+                        &assembly,
```

Comment:
An alternative idea would have been the following:

In appleseed, entities are "parented": you can retrieve the parent of an entity using `get_parent()`. An entity is assigned a parent when it gets inserted into a collection.

What we could have done is manually set the parent of `object_assembly` to be `assembly`. Then `get_or_create_material()` would have been able to walk the chain of parents up until it found the material it was looking for. When `object_assembly` would then be inserted into `assembly.assemblies()`, it's parent would be overwritten (to the same value).

However it would then be unclear in which assembly should `get_or_create_material()` create the material. So all in all not the best solution probably, but I just wanted to share the idea.

Reply:
Hi @dictoon,
I tried suggested approach first, but it didn't work because objects assemblies get added to patent assemblies later. At the point when assembly is populated with materials and objects it doesn't have a parent.

Reply:
Yes, that's why I was suggesting to manually parent the entity first using `set_parent()` but overall I prefer your approach.

Reply:
I tried to parent assembly first and then populate it but it started to crash. I thought it's not designed to be filled out once it's inserted into parent assembly.